analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyToAutoProperty.cs

```diff
@@ -35,158 +35,123 @@ namespace SonarAnalyzer.Rules.CSharp
+            var modifiers = ModifierKinds(accessor1).ToHashSet();
 
-            return accessors.Skip(1).Any(a => !modifiers.SetEquals(GetModifierKinds(a)));
+            return accessors.Skip(1).Any(a => !modifiers.SetEquals(ModifierKinds(a)));
         }
 
-        private static IEnumerable<SyntaxKind> GetModifierKinds(AccessorDeclarationSyntax accessor)
-        {
-            return accessor.Modifiers.Select(m => m.Kind());
-        }
+        private static IEnumerable<SyntaxKind> ModifierKinds(AccessorDeclarationSyntax accessor) =>
+            accessor.Modifiers.Select(m => m.Kind());
 
-        private static bool TryGetFieldFromSetter(AccessorDeclarationSyntax setter, SemanticModel semanticModel, out IFieldSymbol setterField)
+        private static IFieldSymbol FieldFromSetter(AccessorDeclarationSyntax setter, SemanticModel semanticModel)
         {
-            setterField = null;
+            var assignment = AssignmentFromBody(setter.Body) ?? AssignmentFromExpressionBody(setter.ExpressionBody());
 
-            var assignment = GetAssignmentFromBody(setter.Body)
-                ?? GetAssignmentFromExpressionBody(setter.ExpressionBody());
+            return assignment != null
+                   && assignment.IsKind(SyntaxKind.SimpleAssignmentExpression)
+                   && assignment.Right != null
+                   && semanticModel.GetSymbolInfo(assignment.Right).Symbol is IParameterSymbol {Name: "value", IsImplicitlyDeclared: true}
+                   ? FieldSymbol(assignment.Left, semanticModel.GetDeclaredSymbol(setter).ContainingType, semanticModel)
+                   : null;
 
-            if (assignment != null &&
-                assignment.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
-                assignment.Right != null &&
-                semanticModel.GetSymbolInfo(assignment.Right).Symbol is IParameterSymbol parameter &&
-                parameter.Name == "value" &&
-                parameter.IsImplicitlyDeclared)
-            {
-                return TryGetField(assignment.Left, semanticModel.GetDeclaredSymbol(setter).ContainingType,
-                    semanticModel, out setterField);
-            }
-            return false;
-
-            AssignmentExpressionSyntax GetAssignmentFromBody(BlockSyntax body) =>
-                body?.Statements.Count == 1 &&
-                body.Statements[0] is ExpressionStatementSyntax statement
+            AssignmentExpressionSyntax AssignmentFromBody(BlockSyntax body) =>
+                body?.Statements.Count == 1 && body.Statements[0] is ExpressionStatementSyntax statement
                 ? statement.Expression as AssignmentExpressionSyntax
                 : null;
 
-            AssignmentExpressionSyntax GetAssignmentFromExpressionBody(ArrowExpressionClauseSyntax expressionBody) =>
+            AssignmentExpressionSyntax AssignmentFromExpressionBody(ArrowExpressionClauseSyntax expressionBody) =>
```

Comment:
_Line 112 below_: `.ElementAt(0)` should be `.Single()`, becase we tested count=1 before.